### ISSUE TITLE
Handlebars' templates in folders

### ISSUE THREAD

**Issue by displaynone** [nest-modules/mailer#684]
Hi

I want to have the templates organized in folders for different languages, but due to [this](https://github.com/nest-modules/mailer/blob/master/lib/adapters/handlebars.adapter.ts#L38):
```javascript
const templateDir = template.startsWith('./')
  ? lodash_1.get(options, 'dir', '')
  : path.dirname(template);
```
The subfolder is ignored when I configure my template using: `./en/templateName`

Regards

**Comment by Subalee** [comment]
Checkout my comment [here](https://github.com/nest-modules/mailer/issues/570#issuecomment-925927135). It's not completely related but it might help?

**Comment by displaynone** [comment]
Thanks @Subalee but the problem I get is that if I organize folders by language using the same name:
```
/en
  myTemplate.hbs
/es
  myTemplate.hbs
```
Both templates use the same filename (`myTemplate`) so I have to add a prefix to force nest to distinguish them
```
/en
  en_myTemplate.hbs
/es
  es_myTemplate.hbs
```

**Comment by Subalee** [comment]
Hey there, @displaynone 

what I would do in your case is this: 
```
"compilerOptions": {
    "assets": [
      {
        "include": "notification/email-templates/en/**/*.hbs",
        "outDir": "dist/src"
      },
     {
        "include": "notification/email-templates/es/**/*.hbs",
        "outDir": "dist/src"
      }
    ],
    "watchAssets": true
  }
  ```
This builds upon my example but I hope you get what I mean.
The result in the dist should look like this:

`/dist/src/notification/en/myTemplate.hbs`
and
`/dist/src/notification/es/myTemplate.hbs`

Obviously you can change what comes before the localisation to suit your project

This _should_ work in theory, I haven't tested it myself.

**Comment by displaynone** [comment]
Hi again

I have updated to last version and because of this [commit](https://github.com/nest-modules/mailer/commit/9bee48b0e630c71c3702b83747dc41152670947e) we can't use templates in different folders.

The trick you mentioned for `compilerOptions` worked for us, but know it stopped working

If we set the absolute path in the `template`, it removes the subfolders because it takes the dir from the `dir` property in the module:
```javascript
MailerModule.forRoot({
      transport,
      defaults: {
        from: process.env.EMAIL_FROM,
      },
      template: {
        dir: __dirname + '/templates',
```
and we are not able to use relative paths (`./language_folder/`)

**Comment by morpheus-87** [comment]
Should be fixed in `1.7.0`.

**Comment by 2astm** [comment]
don't work for 1.8.1
